mssql-ballerina/Module.md

```diff
@@ -0,0 +1,391 @@
+
+record {|record {} value;|}? result = check resultStream.next();
+
+if result is record {|record {} value;|} {
+    // A valid result is returned.
+    io:println("total students: ", result.value["total"]);
+} else {
+    // The `Student` table must be empty.
+}
+
+error? e = resultStream.close();
+```
+
+#### Updating Data
+
+This sample demonstrates modifying data by executing an `UPDATE` statement via the `execute` remote function of
+the client.
+
+```ballerina
+int age = 23;
+sql:ParameterizedQuery query = `UPDATE students SET name = 'John' 
+                                WHERE age = ${age}`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+#### Deleting Data
+
+This sample demonstrates deleting data by executing a `DELETE` statement via the `execute` remote function of
+the client.
+
+```ballerina
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This sample demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and
+parameterized SQL query as same as the  above `execute` operations.
+
+```ballerina
+// Create the table with the records that need to be inserted.
+var data = [
+  { name: "John", age: 25  },
+  { name: "Peter", age: 24 },
+  { name: "jane", age: 22 }
+];
+
+// Do the batch update by passing the batches.
+sql:ParameterizedQuery[] batch = from var row in data
+                                 select `INSERT INTO students ('name', 'age')
+                                 VALUES (${row.name}, ${row.age})`;
+sql:ExecutionResult[] result = check dbClient->batchExecute(batch);
+```
+
+#### Execute SQL Stored Procedures
+
+This sample demonstrates how to execute a stored procedure with a single `INSERT` statement that is executed via the
+`call` remote function of the client.
+
+```ballerina
+int uid = 10;
+sql:IntegerOutParameter insertId = new;
+
+sql:ProcedureCallResult|sql:Error result = dbClient->call(`exec InsertPerson(${uid}, ${insertId})`);
+if result is error {
+    //An error returned
+} else {
+    stream<record{}, sql:Error>? resultStr = result.queryResult;
+    if resultStr is stream<record{}, sql:Error> {
+        sql:Error? e = resultStr.forEach(function(record{} result) {
+        //can perform operations using 'result'.
+      });
+    }
+    check result.close();
+}
+```
+Note that you have to explicitly invoke the close operation on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
+
+>**Note:** The default thread pool size used in Ballerina is: `the number of processors available * 2`. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
```

Comment:
```suggestion
>**Note:** The default thread pool size used in Ballerina is: `the number of processors available * 2`. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
```
If we are adding this format for the Note here, we need to be consistent and add the same to all the other occurrences as well.